bindings/serialize-v8.cc

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "serialize-v8.h"
+#include "serialize.h"
+
+using namespace v8;
+
+class HeapNode : public Node {
+ private:
+  AllocationProfile::Node *node;
+
+ public:
+  HeapNode(AllocationProfile::Node *node) { this->node = node; }
+
+  virtual std::string name() { return *String::Utf8Value(node->name); }
+
+  virtual std::string filename() {
+    return *String::Utf8Value(node->script_name);
+  }
+
+  virtual int64_t getFileID() { return node->script_id; }
+
+  virtual int64_t lineNumber() { return node->line_number; }
+
+  virtual int64_t columnNumber() { return node->column_number; }
+
+  virtual std::vector<Sample> samples(const std::deque<uint64_t> &stack,
+                                      Profile *p) {
```

Comment:
nit: pointers lean left.

Can you make it a `const Profile* p` to more clearly indicate that `p` is not a return value?

Reply:
I can't use const here -- this function can modify the string table of `Profile* p`